analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxFacade.cs

```diff
@@ -103,15 +103,8 @@ node switch
                 _ => string.Empty
             };
 
-        public override SyntaxNode RemoveConditionalAcesss(SyntaxNode node)
-        {
-            var whenNotNull = node.RemoveParentheses();
-            while (whenNotNull is ConditionalAccessExpressionSyntax conditionalAccess)
-            {
-                whenNotNull = conditionalAccess.WhenNotNull.RemoveParentheses();
-            }
-            return whenNotNull;
-        }
+        public override SyntaxNode RemoveConditionalAccess(SyntaxNode node) =>
+            (node as ExpressionSyntax).RemoveConditionalAccess();
```

Comment:
`node as ExpressionSyntax` can be null, right? isn't it better not to call?

Reply:
Interestingly enough an extension method can be called with `null`, but `RemoveConditionalAccess` itself doesn't handle the null case (we really need to `#nullable enable` the code base at one point).

Reply:
I've looked at it, it's probably handled inside `RemoveParentheses`, but I find this rather risky

Yes, we need `nullable enable` , however until then I'd prefer to stop early for nulls rather then rely on a large call stack (not all our extension methods handle null AFAIK)